overlay-positioner.html

```diff
@@ -0,0 +1,173 @@
+<!--
+    @license
+    Copyright (c) 2015 The Polymer Project Authors. All rights reserved.
+    This code may only be used under the BSD style license found at http://polymer.github.io/LICENSE.txt
+    The complete set of authors may be found at http://polymer.github.io/AUTHORS.txt
+    The complete set of contributors may be found at http://polymer.github.io/CONTRIBUTORS.txt
+    Code distributed by Google as part of the polymer project is also
+    subject to an additional IP rights grant found at http://polymer.github.io/PATENTS.txt
+-->
+
+<link rel="import" href="../polymer/polymer.html">
+
+<!--
+
+Constrains an element's height and width to the window dimensions, and centers it horizontally and
+vertically unless it is already positioned.
+
+-->
+
+<script>
+
+  Polymer.Core = Polymer.Core || {};
+  Polymer.Core.Overlay = Polymer.Core.Overlay || {};
+
+  Polymer.Core.Overlay.Positioner = {
+
+    /**
+     * The element that will be sized with a max-height/width. By default this element is the
+     * target element, but it can be specifically set to a specific element inside the target
+     * if that is more appropriate. This is useful, for example, when a region inside the overlay
+     * should scroll if needed.
+     *
+     * @attribute sizingTarget
+     * @type Node
+     */
+
+    /**
+     * This object memoizes information needed to position and size the target element. By default
+     * the target element is centered along an axis if it is not positioned otherwise. Its size
+     * is constrained to the window size unless a max- or min-width/height is set.
+     *
+     * @attribute dimensions
+     * @type Object
+     */
+
+    get target() {
+      return this;
+    },
+
+    /**
+     * Memoize information needed to position and size the target element.
+     */
+    discoverDimensions: function() {
+      if (this.dimensions) {
+        return;
+      }
+      var target = getComputedStyle(this.target);
+      var sizer = getComputedStyle(this.sizingTarget);
+      this.dimensions = {
+        position: {
+          v: target.top !== 'auto' ? 'top' : (target.bottom !== 'auto' ?
```

Comment:
I'm not sure if this is in our styleguide, but I strongly prefer not abbreviating identifiers. Thoughts?